sway/criteria.c

```diff
@@ -228,8 +228,13 @@ static bool criteria_matches_view(struct criteria *criteria,
 
 		switch (criteria->app_id->match_type) {
 		case PATTERN_FOCUSED:
-			if (focused && strcmp(app_id, view_get_app_id(focused))) {
-				return false;
+			{
+				const char *focused_app_id = view_get_app_id(focused);
```

Comment:
I was tempted to use `xdg_shell_view_from_view` instead to determine the view type, but apparently it's not done this way. I've copied what l.224 is doing.

Reply:
Using `lenient_strcmp` instead